modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -592,10 +592,21 @@ public Map<String, ModelsMap> postProcessAllModels(Map<String, ModelsMap> objs)
         return objs;
     }
 
+    /**
+     * A property is new if it is in a derived class and the derived property is different.
+     * This usually occurs when the data type is different.
+     * We can also consider discriminators as new because the derived class discriminator will have to be defined again
+     * to contain a new value. Doing so prevents having to include the discriminator in the constructor.
+     * @param model
+     * @param property
+     * @return
+     */
     private boolean codegenPropertyIsNew(CodegenModel model, CodegenProperty property) {
```

Comment:
maybe we should rename it to something like codegenPropertyIsNewOrDiscriminator but that's pretty minor.

fyi, later i may refactor this function into codegen model/property class instead.